src/ui/LysisUI.java

```diff
@@ -0,0 +1,249 @@
+
+            if (result != f.APPROVE_OPTION)
+            {
+                return; // User canceled saving the file
+            }
+
+            File file = f.getSelectedFile();
+
+            try
+            {
+                PrintWriter writer = new PrintWriter(new FileOutputStream(file.toString(), false));
+
+                String[] data = LysisUI.this.output.getText().split("\\r?\\n");
+
+                for (String item : data)
+                {
+                    writer.println(item);
+                }
+                writer.close();
+            }
+            catch (Exception ex)
+            {
+                JOptionPane.showMessageDialog(null, "Error writing to file!", "Error!", JOptionPane.ERROR_MESSAGE);
+                return;
+            }
+        }
+    }
+
+    private class LoadFileListener implements ActionListener
+    {
+        @Override
+        public void actionPerformed(ActionEvent e)
+        {
+            JFileChooser f = new JFileChooser();
+            FileNameExtensionFilter filter = new FileNameExtensionFilter("smx or amxx files (*.smx or *.amxx)", "smx", "amxx");
+            f.setFileSelectionMode(JFileChooser.FILES_ONLY);
+            f.setDialogTitle("Select a file to open");
+            f.setFileFilter(filter);
+            f.showOpenDialog(null);
+
+            File file = f.getSelectedFile();
+            if (file == null)
+            {
+                return; // User canceled saving the file
+            }
+
+            LysisUI.this.path = f.getSelectedFile().toString();
+            LysisUI.this.output.append("< File selected! >");
+        }
+    }
+
+    private class CopyButtonListener implements ActionListener
+    {
+        public void actionPerformed(ActionEvent e)
+        {
+            if (!decompiled)
+            {
+                JOptionPane.showMessageDialog(null, "You must first decompile before you copy!", "Error!", JOptionPane.ERROR_MESSAGE);
+                return;
+            }
+
+            Toolkit.getDefaultToolkit().getSystemClipboard().setContents(new StringSelection(output.getText()), null);
+            JOptionPane.showMessageDialog(null, "Copied Successfully!", "Text Copied!", JOptionPane.INFORMATION_MESSAGE);
+        }
+    }
+
+    private class DecompileButtonListener implements ActionListener
+    {
+        public void actionPerformed(ActionEvent e)
+        {
+            if (LysisUI.this.path == null)
+            {
+                JOptionPane.showMessageDialog(null, "Error: No file selected!", "Error!", JOptionPane.ERROR_MESSAGE);
+                return;
+            }
+
+            try
+            {
+                LysisUI.this.output.setText(""); // clear
+
+                File output = File.createTempFile("output", ".tmp");
+                File errors = File.createTempFile("errors", ".tmp");
+
+                Lysis.decompile(new FileOutputStream(output), new FileOutputStream(errors), LysisUI.this.path);
```

Comment:
One thing I want you to check is if creating a temp file for output streams like this is okay. I needed to create an output stream to input into the decompile method and this seemed like the best way, but I'm still a little unsure.